### PR TITLE
externalconn: register azure as a supported External Connection

### PR DESCRIPTION
**Pull request by adityamaru**
`azure` URIs can now be represented using External Connections.

Release note (sql change): Users can now
`CREATE EXTERNAL CONNECTION` to represent an `azure` URI.

Release justification: low risk change to new functionality that registers azure as a supported External Connection provider

### CODE REVIEW

File: pkg/cloud/azure/azure_storage.go [hunk 1]
```diff
@@ -39,20 +39,30 @@ const (
 	AzureAccountKeyParam = "AZURE_ACCOUNT_KEY"
 	// AzureEnvironmentKeyParam is the query parameter for the environment name in an azure URI.
 	AzureEnvironmentKeyParam = "AZURE_ENVIRONMENT"
+
+	scheme = "azure"
```
Comment (benbardin) [inline pkg/cloud/azure/azure_storage.go:43]: Do we want to differentiate between azure-kms and azure-storage?

Comment (adityamaru) [reply pkg/cloud/azure/azure_storage.go:43]: We could, I'm defaulting to the existing schemes that we support if there are no conflicts but we could start moving towards `azure-storage` and `azure-kms`. I'll make the change.

Comment (adityamaru) [reply pkg/cloud/azure/azure_storage.go]: done!

**Review by benbardin — APPROVED** [review]

**Comment by adityamaru** [comment]
`BenchmarkCatchUpScan` seems to have timed out, can't see how that's related to this PR.

bors r=benbardin

**Pull request merged**